Bottom: d5c988584ef608b1811c5e2dd4d96c74a9faf96c
Top:    ae34c573eea6c9bd4a45ef136763597a2929f1ee
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2016-01-17 13:00:00 +0000

drc: Fix polygon-polygon clearance checks

IsPolygonInPolygon was applying 2x bloat to the line it uses to test with
against the second polygon, and IsLineInPolygon was applying a further
bloat offset - making the clearance used incorrect.


---

diff --git a/src/find.c b/src/find.c
index e1c1ee2..31034e5 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2504,7 +2504,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
